test/native/testRunner.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "testRunner.hpp"
+#include <chrono>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+int main() {
+    return TestRunner::instance()->runAllTests();
+}
+
+TestRunner* TestRunner::instance() {
+    static TestRunner instance;
+    return &instance;
+}
+
+bool fileReadable(const char* filename) {
+    int fd = open(filename, O_RDONLY);
+    if (fd == -1) {
+        return false;
+    }
+
+    char buf[1];
+    ssize_t r = read(fd, buf, sizeof(buf));
+    close(fd);
+    return r != -1;
+}
+
+int TestRunner::runAllTests() {
+    int passed = 0;
+    int failed = 0;
+    int skipped = 0;
+    int total_assertions = 0;
+    int i = 1;
+    double total_duration = 0;
+    const int total_tests = testCases().size();
+
+    const bool redirected = !isatty(fileno(stdout));
+    const char* red = redirected ? "" : "\033[31m";
+    const char* green = redirected ? "" : "\033[32m";
+    const char* yellow = redirected ? "" : "\033[33m";
+    const char* reset = redirected ? "" : "\033[0m";
+
+    bool has_only = false;
+    for (auto& pair : testCases()) {
+        has_only |= pair.second.only;
+    }
+    if (has_only) {
+        printf("Only running tests marked with ONLY_TEST_CASE.\n");
+    }
+
+    for (auto& pair : testCases()) {
+        auto& test_case = pair.second;
+        bool force_skip = has_only && !test_case.only;
+
+        printf("Running %s @ %s:%d\n", test_case.name.c_str(), test_case.filename.c_str(), test_case.line_no);
+
+        std::chrono::high_resolution_clock::time_point start = std::chrono::high_resolution_clock::now();
+        if (!force_skip) {
+            test_case.test_function();
+        } else {
+            test_case.skipped = true;
+        }
+        std::chrono::duration<double, std::milli> duration = std::chrono::high_resolution_clock::now() - start;
```

Comment:
IIUC, `std::chrono::high_resolution_clock` is not required to be monotonic and therefore is not suitable for measuring operation duration.
`OS::nanotime()` available in the profiler itself would be a simpler and better alternative.